src/main/java/com/surabhi/taskapp/MyUserDetailsService.java

```diff
@@ -0,0 +1,85 @@
+package com.surabhi.taskapp;
+
+import com.surabhi.taskapp.dto.User;
+import com.surabhi.taskapp.entity.UserEntity;
+import com.surabhi.taskapp.repository.UserRepository;
+import com.surabhi.taskapp.response.Response;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+
+@Slf4j
+@Service
+public class MyUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+
+    public MyUserDetailsService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        log.info("operation =loadUserByUsername, status = IN_PROGRESS, message = load user by Username");
+        UserEntity user = userRepository.findByEmail(email);
+
+        if (user == null) {
+            log.error("operation =loadUserByUsername status = ERROR, msg = error in loadUserByUsername");
+            throw new UsernameNotFoundException("User not found with username: " + email);
```

Comment:
change User not found with username
to User not found with userEmail

Reply:
done